src/units/drawer.hpp

```diff
@@ -60,6 +60,7 @@ class unit_drawer
 	map_location sel_hex;
 	map_location mouse_hex;
 	double zoom_factor;
+	std::set<map_location> units_that_can_reach_goal;
```

Comment:
Is it intentional that `unit_drawer` has its own *copy* of `units_that_can_reach_goal` that doesn't update when the parent set does?

Reply:
That is what I intended to implement, yes, but I'm not sure if that's the best approach.

I tested with this diff:

```
diff --git a/data/scenario-test.cfg b/data/scenario-test.cfg
index c88922bdf40..280e34d5d38 100644
--- a/data/scenario-test.cfg
+++ b/data/scenario-test.cfg
@@ -1505,7 +1505,16 @@ My best advancement costs $next_cost gold and I’m $experience|% there."
                     { x = 23, y = 7, text = string.format("Move cost: %d", c) })
             >>
         [/lua]
+        [kill]
+            side=1,2,3
+            canrecruit=no
+        [/kill]
+        {NOTRAIT_UNIT 2 (Orcish Grunt) 16 4}
+        {NOTRAIT_UNIT 2 (Orcish Grunt) 16 5}
+        {NOTRAIT_UNIT 2 (Orcish Grunt) 16 6}
+        {NOTRAIT_UNIT 2 (Orcish Grunt) 16 7}
+        {NOTRAIT_UNIT 2 (Orcish Grunt) 16 8}
     [/event]
 
     [event]
```

I typed Ctrl+V to enter Show Enemy Moves mode and then moved the mouse among hexes that different numbers of orcs could reach and pressed Ctrl+E to toggle ellipses, which re-runs `unit_drawer::redraw_unit`. I tried this twice, once with the PR as it stands and once with `unit_drawer::redraw_unit` modified to call `game_display::units_that_can_reach_goal` instead of accessing its own data member, and saw no difference between the two attempts.